manufacturing-client/src/main.rs

```diff
@@ -324,11 +325,23 @@ async fn main() -> Result<()> {
                 url = args.manufacturing_server_url;
 
                 mfg_string_type = args.mfg_string_type;
-                // ensure that we are given the iface if MACAddress is selected
-                if mfg_string_type == MfgStringType::MACAddress && args.iface.is_none() {
-                    bail!("When using MACAddress as the MfgStringType --iface is required");
+                if mfg_string_type == MfgStringType::MACAddress {
+                    // user provided iface
+                    if args.iface.is_some() {
+                        iface = args.iface;
+                    } else {
+                        // If user has not selected any specific iface then default iface will be used
+                        match get_default_network_iface() {
+                            Ok(result) => {
+                                iface = result;
+                                log::info!("Default network interface found: {:#?}", iface);
```

Comment:
nit: can you change these to `{variable:#?}`, I know that it is a pain but we had clippy raise warnings before